### ISSUE TITLE
Asynchonous Entity Add When Generating Tree in the Wrong Place

### ISSUE THREAD

**Issue by Seggan** [PaperSpigot/Paper#7458]
### Stack trace

```
[14:10:19 ERROR]: Remove callback for map threw exception 
java.util.concurrent.CompletionException: net.minecraft.ReportedException: Exception generating new chunk
        at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314) ~[?:?]
        at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319) ~[?:?]
        at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1155) ~[?:?]
        at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[?:?]
        at net.minecraft.server.level.ChunkTaskPriorityQueueSorter.lambda$message$1(ChunkTaskPriorityQueueSorter.java:58) ~[app:?]
        at net.minecraft.util.thread.ProcessorMailbox.pollTask(ProcessorMailbox.java:87) ~[app:?]
        at net.minecraft.util.thread.ProcessorMailbox.pollUntil(ProcessorMailbox.java:141) ~[app:?]
        at net.minecraft.util.thread.ProcessorMailbox.run(ProcessorMailbox.java:96) ~[app:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[?:?]
        at java.lang.Thread.run(Thread.java:831) ~[?:?]
Caused by: net.minecraft.ReportedException: Exception generating new chunk
        at net.minecraft.server.level.ChunkTracker.lambda$scheduleChunkGeneration$38(ChunkTracker.java:1340) ~[app:?]
        at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[patched_1.17.1.jar:git-Paper-408]
        at net.minecraft.server.level.ChunkTracker.lambda$scheduleChunkGeneration$40(ChunkTracker.java:1324) ~[app:?]
        at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1146) ~[?:?]
        ... 8 more
Caused by: java.lang.IllegalStateException: Asynchronous entity add!
        at org.spigotmc.AsyncCatcher.catchOp(AsyncCatcher.java:16) ~[patched_1.17.1.jar:git-Paper-408]
        at net.minecraft.server.level.ServerLevel.addEntity0(ServerLevel.java:1365) ~[app:?]
        at net.minecraft.server.level.ServerLevel.addFreshEntity(ServerLevel.java:1306) ~[app:?]
        at net.minecraft.world.level.block.Block.popResource(Block.java:404) ~[patched_1.17.1.jar:git-Paper-408]
        at net.minecraft.world.level.block.Block.popResource(Block.java:370) ~[patched_1.17.1.jar:git-Paper-408]
        at net.minecraft.world.level.block.Block.lambda$dropResources$4(Block.java:355) ~[patched_1.17.1.jar:git-Paper-408]
        at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
        at net.minecraft.world.level.block.Block.dropResources(Block.java:354) ~[patched_1.17.1.jar:git-Paper-408]
        at net.minecraft.server.level.WorldGenRegion.destroyBlock(WorldGenRegion.java:241) ~[app:?]
        at net.minecraft.world.level.LevelWriter.destroyBlock(LevelWriter.java:23) ~[app:?]
        at net.minecraft.world.level.LevelWriter.destroyBlock(LevelWriter.java:19) ~[app:?]
        at net.minecraft.world.level.levelgen.feature.HugeFungusFeature.placeStem(HugeFungusFeature.java:88) ~[app:?]
        at net.minecraft.world.level.levelgen.feature.HugeFungusFeature.place(HugeFungusFeature.java:60) ~[app:?]
        at org.bukkit.craftbukkit.v1_17_R1.CraftRegionAccessor.generateTree(CraftRegionAccessor.java:362) ~[patched_1.17.1.jar:git-Paper-408]
        at org.bukkit.craftbukkit.v1_17_R1.CraftRegionAccessor.generateTree(CraftRegionAccessor.java:276) ~[patched_1.17.1.jar:git-Paper-408]
        at org.bukkit.craftbukkit.v1_17_R1.generator.CraftLimitedRegion.generateTree(CraftLimitedRegion.java:181) ~[patched_1.17.1.jar:git-Paper-408]
        at io.github.addoncommunity.galactifun.base.universe.saturn.Titan$1.populate(Titan.java:162) ~[Galactifun - DEV 58 (git c9b57).jar:?]
        at net.minecraft.world.level.chunk.ChunkGenerator.addDecorations(ChunkGenerator.java:274) ~[patched_1.17.1.jar:git-Paper-408]
        at org.bukkit.craftbukkit.v1_17_R1.generator.CustomChunkGenerator.applyBiomeDecoration(CustomChunkGenerator.java:272) ~[patched_1.17.1.jar:git-Paper-408]
        at net.minecraft.world.level.chunk.ChunkStatus.lambda$static$10(ChunkStatus.java:93) ~[patched_1.17.1.jar:git-Paper-408]
        at net.minecraft.world.level.chunk.ChunkStatus.generate(ChunkStatus.java:212) ~[patched_1.17.1.jar:git-Paper-408]
        at net.minecraft.server.level.ChunkTracker.lambda$scheduleChunkGeneration$38(ChunkTracker.java:1326) ~[app:?]
        at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[patched_1.17.1.jar:git-Paper-408]
        at net.minecraft.server.level.ChunkTracker.lambda$scheduleChunkGeneration$40(ChunkTracker.java:1324) ~[app:?]
        at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1146) ~[?:?]
        ... 8 more
```

### Plugin and Datapack List

Slimefun
Galactifun
I was not given any other info in the original report

### Paper version

Paper 408 1.17.1

### Other

I am not the one who had this, see Slimefun-Addon-Community/Galactifun#116

**Comment by lynxplay** [comment]
Pretty much the same issue as #7001.

Upstream "fixed" that by just not dropping it at all afaik in 1.18, we adopted it in https://github.com/PaperMC/Paper/commit/05a2271884c96f91ecee5f6252660ebdf4d291e1.

**Comment by Seggan** [comment]
We need a fix, not a "fix" 😃

**Comment by Seggan** [comment]
Though if that's enough for you, just say so and I'll tell my client to update MC.

**Comment by lynxplay** [comment]
Well concerning that we did close the original issue after the upstream merge I am inclined to believe this right now works "well enough" for 1.18 and a proper fix that allows the block to be dropped at least has no open PR rn.

If someone is willing to backport upstreams fix to 1.17 is a different question.

**Comment by electronicboy** [comment]
was closed cos the crash was resolved, not that anybody really looked into it much, the real bug which caused that from what jmp said and I just double confirmed that basically the wrong feature is being used for the worlds during that phase
